sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -65,7 +65,8 @@ object SQLExecution {
    */
   def withNewExecutionId[T](
       queryExecution: QueryExecution,
-      name: Option[String] = None)(body: => T): T = queryExecution.sparkSession.withActive {
+      name: Option[String] = None,
+      error: Option[Throwable] = None)(body: => T): T = queryExecution.sparkSession.withActive {
```

Comment:
why is it useful to have this new parameter?

Reply:
To skip getting physical plan info

Reply:
But the `body` will throw an exception and then be caught later. Why do we have this duplicated exception in the parameter?

Reply:
`body` executes later than the start event

Reply:
hmm, I think `queryExecution.executedPlan` will fail and the start event won't be sent. It seems a simpler fix is to update `SQLAppStatusListener` to accept `SparkListenerSQLExecutionEnd` without the start event, which means the query failed at analysis/planning time.

Reply:
I was contemplating the option of sending the end event directly. I don't because the start event could contain the necessary information for the analysis failure reason. e.g., a modified configuration

Reply:
OK, then shall we update the `finally` branch to send the start event as well if it was not sent before? I feel it's awkward to ask the caller side to indicate if it will fail the analysis or not. `def withNewExecutionId` should be smarter about it.

Reply:
Updated to wrap planInfo with `try-catch`